src/components/modals/save-form.component.tsx

```diff
@@ -122,33 +126,55 @@ const SaveForm: React.FC<SaveFormModalProps> = ({ form, schema }) => {
     } else {
       try {
         if (form?.resources?.length != 0) {
-          const valueReference = form?.resources[0].valueReference ?? "";
+          const existingValueReferenceUuid = form?.resources?.find(
+            ({ name }) => name === "JSON schema"
+          )?.valueReference;
 
-          deleteClobdata(valueReference)
+          deleteClobdata(existingValueReferenceUuid)
             .catch((error) =>
               console.error("Unable to delete clobdata: ", error)
             )
             .then(() => {
-              deleteResource(form?.uuid, form?.resources[0].uuid).then(() => {
-                uploadSchema(schema)
-                  .then((res) => {
-                    getResourceUuid(form?.uuid, res.toString).catch((error) =>
-                      console.error("Unable to get resource uuid: ", error)
+              const resourceUuidToDelete = form?.resources?.find(
```

Comment:
This was the key change - when updating a form resource, the backend attempts to infer the resource data type (which doesn't exist in the classpath). The consequence of this is that a runtime exception occurs, which ends up being the root cause of the update operation failing (@samuelmale can probably word this a lot better).